File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*; 
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        Map<String , Integer> Item = new HashMap<>();
        //int days = 0 ;
        for(int i  = 0 ; i<want.length ; i++){
           // days += number[i]; 
            Item.put(want[i] , Item.getOrDefault(want[i] , 0) +number[i]);
        }
       
        int i = 0 ;
        int range = 10;
        int count = 0 ;
            while(range<=discount.length){
            Map<String, Integer>compareItem  = new HashMap<>();
        
            for(int j = i; j<range ; j++){
                if (!Item.containsKey(discount[j])) {
                    break;
                }
                compareItem.put(discount[j] , compareItem.getOrDefault(discount[j] , 0)+1);
            }

            if(Item.equals(compareItem)){
                count++;
            }

            i++;
            range++;

        }
    return count;}
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    for(let i = 0 ; i<want.length ; i++){
        obj[want[i]] = number[i] ; 
    }
    let sum = number.reduce((item , cur) => item + cur) ;
    for(let i = 0 ; i<=discount.length - sum ; i++){
        let newObj = Object.assign({} ,obj) ; 
 
        let slice = discount.slice(i , sum+i) ;
        for(let str of slice){ 
            newObj[str] = newObj[str] -1 ;  
        }      
        let set = new Set(Object.values(newObj)) ;
       ;
        if(set.size === 1 && set.has(0)){
            answer ++ ; 
        }
    } 
   
    return answer;
}


let obj =  {
    
    
   
}